packages/core/src/controllers/relayer.ts

```diff
@@ -392,8 +438,9 @@ export class Relayer extends IRelayer {
   }
 
   private async toEstablishConnection() {
-    if (this.connected) return;
-    if (this.connecting) {
+    await this.confirmOnlineStateOrThrow();
+    if (this.connected && (await isOnline())) return;
```

Comment:
Seems like this is duplicating the `isOnline` check done inside `confirmOnlineStateOrThrow` one line above?